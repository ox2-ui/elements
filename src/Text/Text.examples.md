### Text

    <Text
      customStyle={{ fontSize: '25px' }}
      value="My text"
    />
