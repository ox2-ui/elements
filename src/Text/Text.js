import React, { PropTypes } from 'react';

/**
 * Text Component
 */
const Text = ({ className, style, value, customStyle }) => (
  <div
    className={className}
    style={{ ...style, ...customStyle }}
  >{value}</div>
);


Text.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Text specific style properties.
   */
  customStyle: PropTypes.object.isRequired,
  /**
   * Override the inline-styles of the text element.
   */
  style: PropTypes.object,
  /**
   * Text to display.
   */
  value: PropTypes.string.isRequired,
};

export default Text;
