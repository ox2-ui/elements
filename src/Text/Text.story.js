import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Text from './Text';

storiesOf('Text', module)
  .add('default', () => (
    <Text
      customStyle={{ fontSize: '25px' }}
      value="My text"
    />
  ));
