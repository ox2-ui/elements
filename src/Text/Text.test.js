/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Text from './Text';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Text
      customStyle={{ fontSize: '25px' }}
      value="My text"
    />
  );

  expect(wrapper).toMatchSnapshot();
});
