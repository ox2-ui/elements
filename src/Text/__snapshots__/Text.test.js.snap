// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  customStyle={
    Object {
      "fontSize": "25px",
    }
  }
  value="My text"
>
  <div
    style={
      Object {
        "fontSize": "25px",
      }
    }
  >
    My text
  </div>
</Text>
`;
